### PR TITLE
Unit tests - surface external requests

### PR DESCRIPTION
**Pull request by mokimo**
## Description
**Log if a unit test does a request to an external resource.** 

## Relations
A step towards: [MWPW-129653](https://jira.corp.adobe.com/browse/MWPW-129653)

## Benefits
This takes up bandwidth, leads to slower unit tests and worst of all sporadically fails them if the resource isn't provided on time.
This should provide some feedback to devs writing unit tests and discourage requesting any resource that does not live on localhost.

## Example
Screenshot 2023-05-31 at 14 07 09

### CODE REVIEW

File: web-test-runner.config.mjs [hunk 1]
```diff
@@ -39,4 +39,35 @@ export default {
...
+              );
+            }
+            return oldFetch.call(window, resource, options);
+          };
+          const oldXHROpen = XMLHttpRequest.prototype.open;
+          XMLHttpRequest.prototype.open = async function (...args) {
+            let [method, url, asyn] = args;
+            if (!resource.startsWith("/") && url.startsWith('http://localhost')) {
+              console.error(
+                '** XMLHttpRequest request is disallowed in unit tests, please find a way to mock!',
```
[the remark this answers is not in the document]
Comment (mokimo) [reply web-test-runner.config.mjs]: Makes sense, added!

File: web-test-runner.config.mjs [hunk 2]
```diff
@@ -39,4 +39,35 @@ export default {
     defaultReporter({ reportTestResults: true, reportTestProgress: true }),
     customReporter(),
   ],
+  testRunnerHtml: (testFramework) => `
+    <html>
+      <head>
+        <script type="module">
+          const oldFetch = window.fetch;
+          window.fetch = async (resource, options) => {
+            if (!resource.startsWith("/") && !resource.startsWith('http://localhost')) {
```
Comment (narcis-radu) [inline web-test-runner.config.mjs:48]: nit: use single quotes

File: web-test-runner.config.mjs [hunk 3]
```diff
@@ -39,4 +39,35 @@ export default {
...
+                '** fetch request for an external resource is disallowed in unit tests, please find a way to mock!',
+                resource
+              );
+            }
+            return oldFetch.call(window, resource, options);
+          };
+          const oldXHROpen = XMLHttpRequest.prototype.open;
+          XMLHttpRequest.prototype.open = async function (...args) {
+            let [method, url, asyn] = args;
+            if (!resource.startsWith("/") && url.startsWith('http://localhost')) {
```
Comment (narcis-radu) [inline web-test-runner.config.mjs:59]: single vs double quotes

File: web-test-runner.config.mjs (hunk 2 above)
Comment (mokimo) [reply web-test-runner.config.mjs]: Added!

**Review by narcis-radu — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Pull request merged**